partner_event/models/event_registration.py

```diff
@@ -40,9 +40,14 @@ def _update_attendee_partner_id(self, vals):
             Event = self.env["event.event"]
             # Look for a partner with that email
             email = vals.get("email").replace("%", "").replace("_", "\\_")
-            attendee_partner = Partner.search(
-                [("email", "=ilike", email)], limit=1, order="id"
-            )
+            domain = [("email", "=ilike", email)]
+            partner_id = vals.get("partner_id")
+            if partner_id:
+                commercial_partner = Partner.browse(partner_id).commercial_partner_id
+                domain = [
+                    ("commercial_partner_id", "=", commercial_partner.id)
+                ] + domain
+            attendee_partner = Partner.search(domain, limit=1, order="id")
```

Comment:
What happens if the partner belongs to another entity, you won't reach it...

If you want to prioritize, you should at least search again with no commercial_partner restriction

Reply:
Thanks!